flow/src/main/java/com/sequenceiq/flow/repository/FlowChainLogRepository.java

```diff
@@ -28,4 +32,22 @@ public interface FlowChainLogRepository extends CrudRepository<FlowChainLog, Lon
             + "WHERE fch.flowChainId NOT IN ( SELECT DISTINCT fl.flowChainId FROM FlowLog fl )"
             + " AND fch.flowChainId NOT IN (SELECT DISTINCT fc.parentFlowChainId FROM FlowChainLog fc)")
     int purgeOrphanFlowChainLogs();
+
+    @Query(value = "SELECT fcl.* " +
+            "FROM FlowChainLog fcl " +
+            "INNER JOIN ( " +
+            "    SELECT flowChainId, max(created) as created " +
+            "    FROM FlowChainLog " +
+            "    WHERE flowChainId IN (:flowChainIds) " +
+            "    GROUP by flowChainId " +
+            ")fch on fcl.flowChainId = fch.flowChainId and fcl.created = fch.created", nativeQuery = true,
+            countQuery = "SELECT count(fcl.*) " +
+                    "FROM FlowChainLog fcl " +
+                    "INNER JOIN ( " +
+                    "    SELECT flowChainId, max(created) as created " +
+                    "    FROM FlowChainLog " +
+                    "    WHERE flowChainId IN (:flowChainIds) " +
+                    "    GROUP by flowChainId " +
+                    ")fch on fcl.flowChainId = fch.flowChainId and fcl.created = fch.created")
```

Comment:
this part of the string is duplicated, maybe worth to extract it into a private static variable if we need to change it

Reply:
this is an interface and i can't use a private static String here

Reply:
sorry, just simple String would work, like here: https://github.com/hortonworks/cloudbreak/blob/1ac75096f7bbb82523b713fd1cead6647d591e74/core/src/main/java/com/sequenceiq/cloudbreak/repository/StackDtoRepository.java#L22

Reply:
done